scripts/test_schemas.sh

```diff
@@ -5,21 +5,49 @@ validator=$(docker run -d -p 5001:5000 onsdigital/eq-schema-validator)
 
 sleep 3
 
-echo "Testing Schemas"
+checks=4
+until [ "$checks" == 0 ]; do
+    response="$(curl -so /dev/null -w '%{http_code}' http://localhost:5001/status)"
+
+    if [ "$response" != "200" ]; then
+        echo "$(tput setaf 1)---Error: Schema Validator Not Reachable---"
+        echo "HTTP Status: $response"
+        if [ "$checks"  != 1 ]; then
+            echo -e "Retrying...\\n"
+            sleep 10
+        else
+            echo -e "Exiting...\\n"
+            exit
+        fi
+        tput sgr0
+        (( checks-- ))
+    else
+        (( checks=0 ))
+    fi
+
+done  
 
 exit=0
 
-for schema in $(find $1 -name '*.json');
-do
-echo $schema
-result=$(curl -X POST -H "Content-Type: text/plain" --data "$(cat $schema)" http://localhost:5001/validate 2>/dev/null)
-if [ "$result" != "{}" ] && [ "$result" != "" ]; then
-echo "---Schema Failed Validation---"
-echo "Error: [$result]"
-exit=1
-fi
+echo "---Testing Schemas---"
+for schema in $(find "$1" -name '*.json'); do
+    echo -e "\\nSchema: $schema"
+    result_body=$(curl -s -X POST -H "Content-Type: application/json" -d @"$schema" http://localhost:5001/validate)
+    result_response=$(curl -so /dev/null -w '%{http_code}' -X POST -H "Content-Type: application/json" -d @"$schema" http://localhost:5001/validate)
```

Comment:
Are we calling validate twice for each schema?

Reply:
Grep it is.